R/write_timeseries_dsg.R

```diff
@@ -66,79 +70,102 @@ write_timeseries_dsg = function(nc_file, station_names, lats, lons, times, data,
 		stop('All the collumns in the input dataframe must be of the same type.')
 	}
 	
-	#Lay the foundation. This is a point featureType. Which has one dimension, "obs"
-	#obs_dim = ncdim_def('obs', '', 1:n, unlim = TRUE, create_dimvar=FALSE)
-	station_dim = ncdim_def('station', '', 1:n, create_dimvar=FALSE)
-	time_dim = ncdim_def('time', '', 1:nt, unlim=FALSE, create_dimvar=FALSE)
-	strlen_dim = ncdim_def('name_strlen', '', 1:max(sapply(station_names, nchar)), create_dimvar=FALSE)
-	
-	#Setup our spatial and time info
-	station_var = ncvar_def('station_name', '', dim=list(strlen_dim, station_dim), missval=NULL, prec='char', longname='Station Names')
-	time_var 		= ncvar_def('time','days since 1970-01-01 00:00:00', dim=time_dim, -999, prec='double', longname='time of measurement')
-	lat_var 		= ncvar_def('lat', 'degrees_north', dim=station_dim, -999, prec='double', longname = 'latitude of the observation')
-	lon_var 		= ncvar_def('lon', 'degrees_east', dim=station_dim, -999, prec='double', longname = 'longitude of the observation')
-
-	if(!is.na(alts[1])){
-		alt_var = ncvar_def('alt', 'm', dim=station_dim, missval=-999, prec='double', longname='vertical distance above the surface')
-	}
-	data_vars = list()
+	# Set up data_name var.
 	data_name = data_metadata[['name']]
-	data_vars[[1]] = ncvar_def(data_name, data_unit, dim=list(time_dim, station_dim), prec=data_prec, 
-																longname=data_metadata[['long_name']], missval=-999)
-	if(!is.na(alts[1])){
-		nc = nc_create(nc_file, vars = c(list(lat_var, lon_var, time_var, alt_var, station_var), data_vars))
-	} else {
-		nc = nc_create(nc_file, vars = c(list(lat_var, lon_var, time_var, station_var), data_vars))
-	}
-	nc_close(nc)
-	nc<-nc_open(nc_file, write = TRUE)
-	#add standard_names
-	ncatt_put(nc, 'lat', 'standard_name', 'latitude')
-	ncatt_put(nc, 'time', 'standard_name', 'time')
-	ncatt_put(nc, 'lon', 'standard_name', 'longitude')
 	
-	if(!is.na(alts[1])){
-		ncatt_put(nc, 'alt', 'standard_name', 'height')
+	if(add_to_existing) {
+		# Open existing file.
+		nc<-nc_open(nc_file, write = TRUE)
+		data_vars = list()
+		data_vars[[1]] = ncvar_def(data_name, data_unit, dim=list(nc$dim$time, nc$dim$station), prec=data_prec, 
+															 longname=data_metadata[['long_name']], missval=-999)
+		ncvar_add(nc, data_vars[[1]])
+		nc_close(nc)
+		nc<-nc_open(nc_file, write = TRUE)
+		putDataInNC(nc,nt,n,data_name,data)
+		nc_close(nc)
```

Comment:
This is really just so you can tack on a bunch more variables. Don't want to get carried away right now.